src/logging/LoggingService.ts

```diff
@@ -14,77 +14,38 @@
  * limitations under the License.
  */
 
+import {Property} from '../config/Properties';
 import {DefaultLogger} from './DefaultLogger';
-import {NoLogger} from './NoLogger';
+import {ILogger} from './ILogger';
 
 export enum LogLevel {
+    OFF = -1,
     ERROR = 0,
     WARN = 1,
     INFO = 2,
     DEBUG = 3,
     TRACE = 4,
 }
 
-export interface ILogger {
-    log(level: LogLevel, className: string, message: string, furtherInfo: any): void;
-}
-
 export class LoggingService {
 
-    private static loggingService: LoggingService;
-    logger: ILogger;
-
-    constructor(externalLogger: ILogger = null) {
-        if (externalLogger != null) {
-            this.logger = externalLogger;
-        }
-        if (this.logger == null) {
-            this.logger = new DefaultLogger();
-        }
-    }
-
-    static getLoggingService(): LoggingService {
-        if (LoggingService.loggingService == null) {
-            LoggingService.initialize(null);
-        }
-        return LoggingService.loggingService;
-    }
+    private readonly logger: ILogger;
 
-    static initialize(loggerModule: string | ILogger = null): void {
-        if (typeof loggerModule === 'string') {
-            if (loggerModule === 'off') {
-                LoggingService.loggingService = new LoggingService(new NoLogger());
-            } else if (loggerModule === 'default') {
-                LoggingService.loggingService = new LoggingService();
-            } else {
-                throw new RangeError('Logging type unknown: ' + loggerModule);
-            }
+    constructor(loggingProperty: Property, logLevel: number) {
+        if (loggingProperty == null) {
+            this.logger = new DefaultLogger(logLevel);
+        } else if (this.isLogger(loggingProperty)) {
+            this.logger = loggingProperty;
         } else {
-            LoggingService.loggingService = new LoggingService(loggerModule as ILogger);
+            throw new RangeError('Logging type unknown: ' + loggingProperty);
         }
     }
 
-    log(level: LogLevel, className: string, message: string, furtherInfo: any): void {
-        this.logger.log(level, className, message, furtherInfo);
-    }
-
-    error(className: string, message: string, furtherInfo: any = null): void {
-        this.log(LogLevel.ERROR, className, message, furtherInfo);
-    }
-
-    warn(className: string, message: string, furtherInfo: any = null): void {
-        this.log(LogLevel.WARN, className, message, furtherInfo);
-    }
-
-    info(className: string, message: string, furtherInfo: any = null): void {
-        this.log(LogLevel.INFO, className, message, furtherInfo);
-    }
-
-    debug(className: string, message: string, furtherInfo: any = null): void {
-        this.log(LogLevel.DEBUG, className, message, furtherInfo);
+    isLogger(loggingProperty: Property): loggingProperty is ILogger {
+        return (loggingProperty as ILogger).log !== undefined;
```

Comment:
Is checking one method correct here ? We will be calling `warn` `err` etc on this. They will gave error when used, which will be too late.

Reply:
👍 to sancar